File: c++/InsertionSort.cpp
#include<iostream>
using namespace std;
//int x [5] = {7,3,6,8,5};
void insertionsort(int n , int x[])
{
    int key = 0;

    for(int i = 1; i<n; i++)
    {
        key = x[i];

        for(int j = i-1; j>=0; --j)
        {
            if(x[j] > key)
            {
                x[j+1] = x[j];
                x[j] = key;
            }
        }
    }
}
int main()
{
int n = 5;
int x [5] = {7,4,5,7,3};
insertionsort(n,x);
for(int i = 0; i<n;++i)
{
cout<<x[i]<<" "; 
}
}

File: php/InsertionSort.php
<?php 

class InsertionSort
{
    public $n; 
    public $key;
    public $x = [];

    public function insertion_sort($n , $x)
    {
        $this->n = $n;
        $this->x = $x;

        for($i = 1 ; $i<$this->n;$i++)
        {
            $this->key = $key = $x[$i];

            $j = $i-1; 
            while($j>=0 && $this->key<$this->x[$j])
            {
                $this->x[$j+1] = $this->x[$j];
                $j--;
            }
            $this->x[$j+1] = $key;
        }
    }
}

$sort = new InsertionSort;
$n = 5 ;
$array = [
  0=>7,
  1=>3,
  2=>2,
  3=>9,
  4=>4,
];
$sort->insertion_sort($n,$array);

for($i = 0; $i<$n; $i++)
{
    echo $sort->x[$i]." "; 
}